namespace-provisioner/troubleshooting.hbs.md

```diff
@@ -0,0 +1,136 @@
+
+For example, let's assume that the following file is used as `additional_sources`
+
+```yaml
+#@ load("@ytt:data", "data")
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: gitlab-workload-token
+  annotations:
+    tekton.dev/git-0: https://git.company.com
+type: kubernetes.io/basic-auth
+stringData:
+  #! data.values does not have any key of gl-secret-user nor gl-secret-pass
+  username: #@ data.values.gl-secret-user
+  password: #@ data.values.gl-secret-pass
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: github-workload-token
+  annotations:
+    tekton.dev/git-0: https://github.aetna.com
+type: kubernetes.io/basic-auth
+stringData:
+  #! data.values does not have any key of gh-secret-user nor gh-secret-pass
+  username: #@ data.values.gh-secret-user
+  password: #@ data.values.gh-secret-pass
+---
+```
+
+Where the used `data.values` does not exist, and after adding it as an `additional_source` in the `witherror` library, the Kapp App will show an error as follows:
+
+```bash
+$ kubectl -n tap-namespace-provisioning get app/provisioner --template={{.status.usefulErrorMessage}}
+
+ytt: Error:
+- library.eval: Evaluating library 'witherror':
+    in <toplevel>
+      template.yaml:155 | #@          instances = overlay.apply(instance.eval(), customize())
+
+    reason:
+     - struct has no .gl_secret_user field or method
+         in <toplevel>
+           _ytt_lib/witherror/secrets.yaml:12 |   username: #@ data.values.gl_secret_user
+```
+
+This will show any error coming from the Carvel-YTT template resolution.
+
+Another common error is defining resources several times (like adding a resource which is created as default instead of ovelaying it), that will be reported in the Kapp App as well
+
+### <a id="unable-to-delete-namespace"></a>Unable to delete namespace
+
+When a namespaces is provisioned by be present in the `desired-namespaces` (via controller or GitOps), then it can be deleted in a common way by `kubectl` command
+
+```bash
+kubectl delete namespace {namespace-name}
+```
+
+but it needs to be *cleared* of workload before being deleted.
+
+When the provisioned namespace is deleted and there is a workload already created in there, it is very likely that the *`namespace`* will keep the `Terminating` state because some resources can not be deleted.
+
+- One of the causes of this behavior is because the workload creates a Carvel Kapp App to manage the workload referencing the `ServiceAccount` in the *`namespace`* and the `kubectl` command does not take precedence of the `ServiceAccount` over the Kapp App then it can be deleted first, and then the workload Kapp App will block the namespace termination failing itself while waiting the `ServiceAccount` to exist with a *`finalizer`* (`finalizers.kapp-ctrl.k14s.io/delete`).
```

Comment:
It would be something like this:

The workload creates a Carvel Kapp App that references the `ServiceAccount` in the *`namespace`*. As `kubectl` does not respect Carvel kapp-controller delete order and the  `ServiceAccount` is deleted before the workload Carvel App. As a result, the Carvel App will block the namespace termination while waiting for the `ServiceAccount` to exist with a finalizer (finalizers.kapp-ctrl.k14s.io/delete) messaz